### PR TITLE
MLv2: Fix metadata inference for `:field` clauses with string names that come from joins

### LINKED ISSUE #29902: [MLv2] [Bug] `:field` clauses with string names not working if those Fields come from joins

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #29902

Metadata calculation was erroring for `:field` clauses that used string names and had a `:join-alias`. This PR fixes it

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29903)

### CODE REVIEW

File: src/metabase/lib/stage.cljc [hunk 1]
```diff
@@ -240,17 +241,10 @@
    source-table-id :- [:or ::lib.schema.id/table ::lib.schema.id/table-card-id-string]
    unique-name-fn  :- fn?]
   (when-let [card-id (lib.util/string-table-id->card-id source-table-id)]
-    ;; it seems like in some cases the FE is renaming `:result_metadata` to `:fields`, not 100% sure why but
```
Comment (camsaul) [inline src/metabase/lib/stage.cljc:243]: Moved this to `metabase.lib.card`

File: src/metabase/lib/stage.cljc [hunk 2]
```diff
@@ -28,7 +29,7 @@
   (comp (mbql.u/unique-name-generator
          ;; unique by lower-case name, e.g. `NAME` and `name` => `NAME` and `name_2`
          :name-key-fn     u/lower-case-en
-         ;; truncate alias to 30 characters (actually 21 characters plus a hash).
+         ;; truncate alias to 60 characters (actually 51 characters plus a hash).
```
Comment (camsaul) [inline src/metabase/lib/stage.cljc:32]: 60 is actually the correct number

File: src/metabase/lib/stage.cljc [hunk 3]
```diff
@@ -60,7 +61,7 @@
      lib.metadata.calculation/ColumnMetadataWithSource
      [:map
       [:lib/source-column-alias  ::lib.schema.common/non-blank-string]
-      [:lib/desired-column-alias ::lib.schema.common/non-blank-string]]]]
+      [:lib/desired-column-alias [:string {:min 1, :max 60}]]]]]
```
Comment (camsaul) [inline src/metabase/lib/stage.cljc:64]: desired column alias should never be more than 60 characters

**Review by metamben — APPROVED** [review]
:lgtm:

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29903#-:-NSXbo9OBUtie7hAOy6l:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @camsaul)

___
*[`src/metabase/lib/card.cljc` line 22 at r1](https://reviewable.io/reviews/metabase/metabase/29903#-NSXSxN57jFUHaiO4CH6:-NSXSxN57jFUHaiO4CH7:b-mlis7a) ([raw file](https://github.com/metabase/metabase/blob/02c48a38570f9242b5eb65e19afa68710685286b/src/metabase/lib/card.cljc#L22)):*
> ```Clojure
>   ;; handle that case anyway. (#29739)
>   (when-let [card (lib.metadata/card query card-id)]
>     (when-let [result-metadata (or ((some-fn :result_metadata :fields) card)
> ```

I find
```clojure
(:result_metadata card)
(:fields card)
```
easier to read than



_Code quote:_
```Clojure
((some-fn :result_metadata :fields) card)
```

**Pull request merged**